src/BenchmarkDotNet/Locators/ILocator.cs

```diff
@@ -0,0 +1,10 @@
+using System;
+using System.IO;
+
+namespace BenchmarkDotNet.Locators;
+
+public interface ILocator
+{
+    LocatorType LocatorType { get; }
+    FileInfo Locate(DirectoryInfo rootDirectory, Type type);
```

Comment:
I don't think custom locators need the root directory that BDN uses. They can find the root directory again themselves if they need it. I also think it makes more sense to pass the assembly, rather than the type.

```suggestion
    FileInfo Locate(Assembly assembly);
```

Reply:
If this is intended to be extended in the future, it probably makes sense to pass in a `LocatorArgs` class so it can be updated without breaking changes.